docs/docs/troubleshooting.md

```diff
@@ -0,0 +1,33 @@
+---
+id: troubleshooting
+title: Troubleshooting
+sidebar_title: Troubleshooting
+---
+### Summary
+
+The following page provides suggestions for common errors that may occur during firmware compilation. If the information provided is insufficient to resolve the issue, feel free to seek out help from the [ZMK Discord](https://zmkfirmware.dev/community/discord/invite).
+
+### Environment Variables not Defined
```

Comment:
Thinking through this, I wonder if a header of `### CMake Errors` which pinpoints the user visible error, not the root cause, might help folks identify this as relevant to them sooner.